closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurity.java

```diff
@@ -347,7 +347,7 @@ private static void getProfileID(Properties props) {
             boolean profileExists = false;
             for (Object keyObject : props.keySet()) {
                 if (keyObject instanceof String key) {
-                    if (key.startsWith(potentialProfileID)) {
+                    if (key.startsWith(potentialProfileID + '.')) {
```

Comment:
Please capture `potentialProfileID + '.'` in a local rather than compute it for each key.
```java
            String profilePrefix = potentialProfileID + '.';
            for (Object keyObject : props.keySet()) {
                if (keyObject instanceof String key) {
                    if (key.startsWith(profilePrefix)) {
```

Reply:
Done.